### PR TITLE
Sync process updates

### PR DESCRIPTION
**Pull request by theharq**
#### What this PR does:

- Add project to CodeClimate and update the badge reference in the README
- Disable external requests on tests.
- Remove `rescue` from Exception, _Never rescue from Exception_
- Fix table name reference for sequel model, bad queries generated breaking transactions.
- Fix issue of calling `operation` for nil class, sequel returns nil if there are no changes in the object when calling `update`
- Extract Methods, Replate temp variable with query, Rename methods, Rename variables, etc.
- Make rubocop happy. (style improvements)
- Add more tests
- Delete rake task tariff:sync:download
- Fix flaky specs related to `TariffSynchronizer.host` attribute

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [![Circle CI](https://circleci.com/gh/bitzesty/trade-tariff-backend.svg?style=svg)](https://circleci.com/gh/bitzesty/trade-tariff-backend)
-[![Code Climate](https://codeclimate.com/github/alphagov/trade-tariff-backend/badges/gpa.svg)](https://codeclimate.com/github/alphagov/trade-tariff-backend)
+[![Code Climate](https://codeclimate.com/github/bitzesty/trade-tariff-backend/badges/gpa.svg)](https://codeclimate.com/github/alphagov/trade-tariff-backend)
```
Comment (matthewford) [inline README.md]: realistically alphagov should be the main org that controls this repo for now, even once we move to CF we would be keeping both these repos in sync so, I would revert this change.

Comment (theharq) [inline README.md]: :ok_hand:

**Comment by matthewford** [comment]
As an aside looks like CircleCI needs to be setup for this repo (I can't see the callbacks on this PR).
